gateway/src/main/kotlin/com/example/gateway/web/rest/PassController.kt

```diff
@@ -0,0 +1,85 @@
+package com.example.gateway.web.rest
+
+import com.example.gateway.configuration.NatsClient
+import com.example.gateway.web.dto.PassDto
+import com.example.gateway.web.mapper.proto.pass.CancelPassResponseMapper.toUnitResponse
+import com.example.gateway.web.mapper.proto.pass.CreatePassResponseMapper.toCreatePassRequest
+import com.example.gateway.web.mapper.proto.pass.CreatePassResponseMapper.toDto
+import com.example.gateway.web.mapper.proto.pass.DeletePassByIdResponseMapper.toDeleteResponse
+import com.example.gateway.web.mapper.proto.pass.FindPassByIdResponseMapper.toDto
+import com.example.gateway.web.mapper.proto.pass.TransferPassResponseMapper.toTransferResponse
+import com.example.internal.NatsSubject.Pass.CANCEL
+import com.example.internal.NatsSubject.Pass.CREATE
+import com.example.internal.NatsSubject.Pass.DELETE_BY_ID
+import com.example.internal.NatsSubject.Pass.FIND_BY_ID
+import com.example.internal.NatsSubject.Pass.TRANSFER
+import com.example.internal.input.reqreply.CancelPassRequest
+import com.example.internal.input.reqreply.CancelPassResponse
+import com.example.internal.input.reqreply.CreatePassResponse
+import com.example.internal.input.reqreply.DeletePassByIdRequest
+import com.example.internal.input.reqreply.DeletePassByIdResponse
+import com.example.internal.input.reqreply.FindPassByIdRequest
+import com.example.internal.input.reqreply.FindPassByIdResponse
+import com.example.internal.input.reqreply.TransferPassRequest
+import com.example.internal.input.reqreply.TransferPassResponse
+import jakarta.validation.Valid
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import reactor.core.publisher.Mono
+
+@RestController
+@RequestMapping("/passes")
+internal class PassController(private val natsClient: NatsClient) {
```

Comment:
such dependency is acceptable for now but it will be problem when you reach hexagonal architecture